### ISSUE TITLE
`createStagingRepository` fails when group is only set via `MavenPublication.groupId`

### ISSUE THREAD

**Issue by RBusarow** [vanniktech/gradle-maven-publish-plugin#445]
This is new behavior in `0.22.0`, using the base plugin.  The Gradle version is `7.5.1`, but that shouldn't matter.

I don't have a full stacktrace at the moment but the exception message made it pretty clear what's happening:

> No matching staging profile found in account rbusarow. It is expected that the account contains a staging profile that matches or is the start of workflow.workflow-config.Available profiles are: com.squareup

The `createStagingRepository` task is using `project.group` as the groupId.

I have been setting `groupId` alongside `artifactId` like this:

```kotlin
configure<PublishingExtension> {
  publications.withType(MavenPublication::class.java) pub@{
    this@pub.groupId = "com.example"
    this@pub.artifactId = "some-id"
  }
}
```

This is done in order to an old conflict resolution issue in Gradle (https://github.com/gradle/gradle/issues/847).  To summarize, a unique `project.group` is necessary to distinguish project dependencies when they have the same simple name, like `:a:api` and `:b:api`.

But now with 0.22.0, `MavenPublishBaseExtension` is hard-coded to get the group from the project:

https://github.com/vanniktech/gradle-maven-publish-plugin/blob/2e12d99d80fac3d7b4a0f839757d8fb1ffb5dc5a/plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt#L64-L66

Ideally, the extension and `CreateSonatypeRepositoryTask` would also use the publication's `groupId` value.

**Comment by gabrielittner** [comment]
The reason why we are not using the group id from the publications is that they might be different. For example if you publish Gradle plugins you will get additional publications which use the plugin id as their group id. I will try to come up with a solution for this. My first thought is to also explicitly add group/artifactId/version to our DSL and then when setting it there or in through gradle properties to not modify `project.group` and project.version` but instead directly apply these to the publication.

**Comment by gabrielittner** [comment]
This will be fixed in 0.23.0 and you can already try it out with the snapshots. The plugin will not set `project.group` and `project.version` anymore so you can use the former to differentiate projects. For the published group id we will use either the `GROUP` gradle property or you can pass group, artifact id and version to the new `coordinates` method in our DSL
